### ISSUE TITLE
input to generator function seems to be missing trailing newlines

### ISSUE THREAD

**Issue by tcheinen** [gobanos/cargo-aoc#65]
I ran into some trouble with today's (2020/day 4) problem because the generator was passing a truncated form of the input files sans any trailing newlines.  I'm unclear as to why this happens because my functioning test case also uses include_str! but its replicable and happens to any number of trailing newlines.  I'm using aoc-runner and aoc-runner-derive version 0.30 and cargo-aoc version 0.3.2.

**Comment by gobanos** [comment]
Hum, input is trimmed to ease parsing (for example, when parsing a number on each line) here : https://github.com/gobanos/cargo-aoc/blob/52164cec883489167ad0bd4c6a6229aa588ea123/aoc-runner/src/lib.rs#L17
There is currently no way to turn it off :/

**Comment by tcheinen** [comment]
Oh okay, I missed that while I was looking through the source.  It might be worth mentioning that in the readme (or adding a way to turn it off) since it threw me for a loop this morning.  My parser depended on a trailing newline for each passport and although it was easy enough to fix I didn't realize I needed to do that.

**Comment by tcheinen** [comment]
Actually, what would the use case for that?  `str::lines` will split a string into lines and it gives the same result with and without a trailing newline.

**Comment by gobanos** [comment]
You're right, a flag to prevent trimming could be nice. After a glance at some old inputs, it still seems like a good default to me.

**Comment by aldanor** [comment]
I would go as far as saying it would make sense to not trim anything at all - because otherwise a solution that works without cargo-aoc may not work with it, and vice versa.

**Comment by dnlmlr** [comment]
So I'm pretty much zombifying this issue, but this behavior is absolutely unexpected and should not be the default imo. 

Trimming the input in your solution or a generator function is literally one line, if you actually need it. Efficiently untrimming the input on the other hand is pretty much impossible without bad hacks. 

This breaks for example a possible optimized solution with `array_chunks::<4>()` for aoc2022 day2. All lines consist of a structure like `A X\n`, which is exactly 4 bytes. When the trailing newline is trimmed, this skipps the last line and stops working.
